### PR TITLE
Move specialist topic tagging edition page to the GOV.UK Design System

### PR DESCRIPTION
**Pull request by dnkrj**
⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

## What

[Move specialist topic tagging edition page to the GOV.UK Design System.
](https://trello.com/c/WpB7xxzc/614-move-specialist-topic-tagging-edition-page-to-the-govuk-design-system)

This PR does not address any of the differences between the behaviour of the bootstrap select and the design system autocomplete or include the cancel button as these will be tackled in other tickets.

## Why

This is part of a Whitehall transition work to move from bootstrap to GOVUK design system.

### After

whitehall-admin dev gov uk_government_admin_editions_1372490_legacy_associations_edit(iPad Pro)
whitehall-admin dev gov uk_government_admin_editions_1372490_legacy_associations_edit(iPad Pro) (3)
whitehall-admin dev gov uk_government_admin_editions_1372490_legacy_associations_edit(iPhone 12 Pro) (1)

### Before

whitehall-admin dev gov uk_government_admin_editions_1372490_legacy_associations_edit(iPad Pro) (2)
whitehall-admin dev gov uk_government_admin_editions_1372490_legacy_associations_edit(iPad Pro) (1)
whitehall-admin dev gov uk_government_admin_editions_1372490_legacy_associations_edit(iPhone 12 Pro)

### CODE REVIEW

**Comment by nnagewad** [comment]
Looks good bud!

File: test/functional/edition_legacy_associations_controller_test.rb [hunk 1]
```diff
@@ -3,6 +3,10 @@
 class Admin::EditionLegacyAssociationsControllerTest < ActionController::TestCase
   include Admin::EditionRoutesHelper
 
+  setup do
+    login_as :gds_editor
```
[the remark this answers is not in the document]
Comment (davidgisbey) [reply test/functional/edition_legacy_associations_controller_test.rb:7]: Yea i'd add a new feature spec to the `specialist-sectors.feature` with the permission so there's one with and without.

```
  Scenario: writer can tag documents with specialist sectors
    Given I am a writer
    And there are some specialist sectors
    When I start editing a draft document
    And I continue to the legacy tagging page
    Then I can tag it to some specialist sectors

Scenario: writer can tag documents with specialist sectors with `Preview design system` permission
    Given I am a writer
    And i have the "Preview design system" permission
    And there are some specialist sectors
    When I start editing a draft document
    And I continue to the legacy tagging page
    Then I can tag it to some specialist sectors
```

If there's relevant view tests that need to change due to styling or user interaction you can duplicate them and add the permission to the user

```
@current_user.permissions << "Preview design system"
```

File: app/views/admin/base/_specialist_sector_fields.html.erb [hunk 2]
```diff
@@ -0,0 +1,31 @@
+<p class="govuk-body govuk-!-margin-bottom-7">
+  Do not tag to a specialist topic unless your organisation maintains that topic page. 
+  <a class="govuk-link" href="TODO">Read our blog post</a> for more information.
```
[the remark this answers is not in the document]
Comment (dnkrj) [reply app/views/admin/base/_specialist_sector_fields.html.erb]: I spoke to JP, Nikin and Carlos and no one knew where the link came from so have removed it.

**Review by ollietreend — APPROVED** [review]
Looks good 👍🏻 thanks for adding the feature spec

**Pull request merged**